lib/engine/game/g_18_co.rb

```diff
@@ -150,12 +152,28 @@ def revenue_str(route)
         str
       end
 
+      def upgrades_to?(from, to, special = false)
+        # Green towns can't be upgraded to brown cities unless the hex has the upgrade icon
+        if GREEN_TOWN_TILES.include?(from.hex.tile.name)
+          return BROWN_CITY_TILES.include?(to.name) if from.hex.tile.icons.select { |icon| icon.name == 'upgrade' }.any?
```

Comment:
How is the upgrade icon being set?

Reply:
The upgrade icon is specific to 18CO and set in the map config.

Reply:
icons.any? { ... }

Reply:
Done